tests/parametric/test_parametric_endpoints.py

```diff
@@ -0,0 +1,733 @@
+
+    def test_get_baggage(self, test_agent, test_library):
+        """
+        Validates that /trace/span/get_baggage gets a baggage item from a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - key: str
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.start_span("span") as s1:
+                s1.set_baggage("key", "value")
+
+                baggage = s1.get_baggage("key")
+                assert baggage == "value"
+
+    def test_get_all_baggage(self, test_agent, test_library):
+        """
+        Validates that /trace/span/get_all_baggage gets all baggage items from a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        Supported Return Values:
+        - baggage: Dict[str, str]
+        """
+        with test_library:
+            with test_library.start_span("span") as s1:
+                s1.set_baggage("key1", "value")
+                s1.set_baggage("key2", "value")
+
+                baggage = s1.get_all_baggage()
+                assert baggage["key1"] == "value"
+                assert baggage["key2"] == "value"
+
+    def test_remove_baggage(self, test_agent, test_library):
+        """
+        Validates that /trace/span/remove_baggage removes a baggage item from a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - key: str
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.start_span("span") as s1:
+                s1.set_baggage("key", "value")
+                s1.remove_baggage("key")
+
+            headers = test_library.inject_headers(s1.span_id)
+            assert not any("baggage" in header for header in headers)
+
+    def test_remove_all_baggage(self, test_agent, test_library):
+        """
+        Validates that /trace/span/remove_all_baggage removes all baggage items from a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.start_span("span") as s1:
+                s1.set_baggage("key1", "value")
+                s1.set_baggage("key2", "value")
+                s1.remove_all_baggage()
+
+            headers = test_library.inject_headers(s1.span_id)
+            assert not any("baggage" in header for header in headers)
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_OtelSpan_Start_Finish:
+    def test_span_start_and_finish(self, test_agent, test_library):
+        """
+        Validates the /trace/otel/start_span API creates a new span:
+
+        Supported Parameters:
+        - name: str
+        - timestamp (μs): Optional[int]
+        - span_kind: Optional[Literal[0, 1, 2, 3, 4, 5]] # integers correspoond -> ["UNSPECIFIED", "INTERNAL", "SERVER", "CLIENT", "PRODUCER", "CONSUMER"]
```

Comment:
Instead of passing these as ints, can we instead pass them as strings? Also, UNSPECIFIED should be renamed to INTERNAL based on the OpenTelemetry API: https://opentelemetry.io/docs/specs/otel/trace/api/#spankind

Reply:
Actually my plan is to pass in an Enum ([PR](https://github.com/DataDog/system-tests/pull/3370/files#diff-f5f86992245595c4ea7be47e8ce6502becdc5166459538487b189b106ff00093)). This change involves updating multiple lines so I am planning to make it in a future PR. For now I am just documenting how the endpoints work.